base/strings/string.jl

```diff
@@ -190,14 +191,151 @@ end
 end
 
 ## checking UTF-8 & ACSII validity ##
+#=
+    The UTF-8 Validation is performed by a shift based DFA.
+    Using the state machine diagram found @ https://bjoern.hoehrmann.de/utf-8/decoder/dfa/
+
+        Important States
+            0 -> UTF8_ACCEPT is the start state and represents a complete UTF-8 String as well
+                        ASCII only strings will never leave this state
+            1 -> UTF8_INVALID is only reached by invalid bytes and once in this state will not
```

Comment:
Seems the state description is incomplete

Reply:
I only was documenting the important steps no updated it none the less

Reply:
still an issue